
trycatch = fn (Arg){
    try {
        make_error(Arg)
    } catch (throw 1) {
    	# catch with parenthesis
        io.format("throw 1~n")
    } exit 2 {
    	# catch without parenthesis
        io.format("exit 2~n")
    } error (some fancy tuple 3) {
        io.format("error~n")
    } else {
        io.format("else~n")
    }
}

make_error = fn (1) {
    throw(1)
} (2) {
    exit(2)
} (3) {
    erlang.error((some fancy tuple 3))
}

run = fn () {
    trycatch(1)
    trycatch(2)
    trycatch(3)
}
